src/common/CMakeLists.txt

```diff
@@ -19,6 +19,8 @@ endif()
 
 shared_add_library(sfscommon ${COMMON_SOURCES})
 shared_target_link_libraries(sfscommon ${YAML_CPP_LIBRARIES})
+target_link_libraries(sfscommon sfserr)
+target_link_libraries(sfscommon_pic sfserr_pic)
```

Comment:
I am not sure if this will work for all scenarios. The PIC targets are not always available, they are if the client-library (dynamic) is selected. I also don't see a direct usage of a _pic target like this in other parts of the project.
Please search in all the cmake files by: 'target_link_libraries(' to confirm it.

Reply:
> But I think it works both with shared libraries enabled and disabled.

Nope, you were right. I missed this because enabling tests enables shared libraries.

Reply:
This fixed the issue for both shared and non-shared builds:

`shared_target_link_libraries(sfscommon ${YAML_CPP_LIBRARIES} STATIC sfserr SHARED sfserr_pic)`

However, I expected shared_target_link_libraries to work. I would have expected it to add sfserr and sfserr_pic to the respective libraries sfscommon and sfscommon_pic. I think this might be because `src/common/` is added before `src/common/sfserr/`, thus causing it not to find the library in the shared_target_link_libraries and not adding it in the PIC library.

In any case, I think this is more explicit about it and works regardless of ordering of add_subdirectory.